float32[] pos_x_list
float32[] pos_y_list
float32[] vel_x_list
float32[] vel_y_list